### ISSUE TITLE
ARIA labels, and more accessibility features

### ISSUE THREAD

**Issue by zerline** [jupyter-widgets/ipywidgets#2650]
We should provide computed ARIA attributes wherever needed, and let the user set them programmatically.
- a Button with no description -> could take the `tooltip` value, or any programmatically set value, as its `aria-label` value
- any form input with no label -> could take the `tooltip` or a custom `aria-label`
- set `aria-autocomplete` for Combobox
- set `aria-placeholder` on all controls having a non-empty `placeholder`
- ..
cf https://www.w3.org/TR/wai-aria/#attrs_widgets
